File: daily/2026/1582_Special_Positions_in_a_Binary_Matrix.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        int m = mat.size();
        int n = mat[0].size();
        vector<int> rowsum(m, 0);
        vector<int> colsum(n, 0);
        for (int i = 0; i < m; i++){
            int acc = 0;
            for (int j = 0; j < n; j++){
                acc += mat[i][j];
            }
            rowsum[i] = acc;
        }
        for (int j = 0; j < n; j++){
            int acc = 0;
            for (int i = 0; i < m; i++){
                acc += mat[i][j];
            }
            colsum[j] = acc;
        }
        int ans = 0;
        for (int i = 0; i < m; i++){
            for (int j = 0; j < n; j++){
                if (mat[i][j] == 1 && rowsum[i] == 1 && colsum[j] == 1) ans++;
            }
        }
        return ans;
    }
};

int main(){
    vector<vector<int>> mat = {{1,0,0},{0,0,1},{1,0,0}};
    // vector<vector<int>> mat = {{1,0,0},{0,1,0},{0,0,1}};
    Solution S;
    cout << S.numSpecial(mat) << endl;
    return 0;
}

File: daily_python/1582_Special_Positions_in_a_Binary_Matrix.py
import os
import sys
from typing import List

class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        rowsum = [sum(r) for r in mat]
        colsum = [sum(c) for c in zip(*mat)]
        ans = 0
        for r in range(m):
            for c in range(n):
                if mat[r][c] == 1 and rowsum[r] == 1 and colsum[c] == 1:
                    ans += 1
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            mat = [list(map(int, r.split(','))) for r in lines[0].split('=')[1].strip()[2:-3].split('},{')]
            dataset.append(mat)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for mat in dataset:
        results.append(solution.numSpecial(mat))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')